packages/engine/src/worker/runner/javascript/batch.js

```diff
@@ -183,41 +185,32 @@
     // JS Arrow doesn't really document what `builder.finish` does, but
     // maybe it affects later serialization.
     builder.finish();
-    return builder;
+    return builder.flush();
   };
 
   // TODO: Can JS Arrow silently coerce some flushed values to different types if
   //       their type differs from what it's supposed to be according to the schema?
-  const array_data_from_builder = (builder) => {
-    // TODO: Union arrays.
-    // TODO: Use Arrow Data methods and function `builder.flush` more straightforwardly.
-
-    const len = builder.length;
-    const null_count = builder.nullCount;
-    const null_bits = builder._nulls ? builder._nulls.flush(len).buffer : null;
-    // `.buffer` gets Uint8Array's underlying ArrayBuffer for FFI.
-    // ArrayData offset is always 0. (Not to be confused with offset buffer.)
+  const ffi_data_data_from_array_data = (array_data) => {
```

Comment:
Yeah absolutely, that was a typo anyway

Reply:
Are we okay with `ffi_data_from_array_data` or do you want something else?
I agree it's still somewhat unreadable but it'll be hard to capture the relevant information otherwise.

The `DataFfi` struct is almost a duplicate of the `ArrayData` struct anyway (although annoyingly in JS they just call it `Data` so it'd be even more confusing)

Reply:
Yeah it's alright